meson.build

```diff
@@ -287,23 +287,41 @@ endif
 # Compression support
 #-------------------------------------------------------------------------------
 
-zstd = dependency('libzstd', version : '>= 1.4.4', required : get_option('zstd'))
+zstd_dlopen = get_option('dlopen').contains('zstd')
+if get_option('zstd').disabled() and zstd_dlopen
+  error('Incompatible options: dlopen\'ed zstd needs zstd at build time')
+endif
```

Comment:
For the `dlopen` case we should be using `foo.partial_dependency(compile_args : true)`. As-is we explicitly pass `-lfoobar` and if we get a symbol snafu (don't properly use the wrapper macros) kmod will be linked against the library.

Ideally we'll go with the "just use dlopen" mentioned above, which will remove the build system complexity. Plus we don't risk having a semi-broken setup for the extra build permutations.

Reply:
I guess we should also have `includes : true` in that partial_dependency() call. I'm adding a few more refactors and doing that.